src/pytest_plugins/forks/forks.py

```diff
@@ -204,63 +208,95 @@ def fork(request):
     pass
 
 
+def get_validity_marker_args(
+    metafunc: Metafunc,
+    validity_marker_name: str,
+    test_name: str,
+) -> str | None:
+    """Check and return the arguments specified to validity markers.
+
+    Check that the validity markers:
+
+    - `pytest.mark.valid_from`
+    - `pytest.mark.valid_until`
+    - `pytest.mark.valid_at_transition_to`
+
+    are applied at most once and have been provided with exactly one
+    argument which is a valid fork name.
+
+    Args:
+        metafunc: Pytest's metafunc object.
+        validity_marker_name: Name of the validity marker to validate
+            and return.
+        test_name: The name of the test being parametrized by
+            `pytest_generate_tests`.
+
+    Returns:
+        The name of the fork specified to the validity marker.
+    """
+    validity_markers = [
+        marker for marker in metafunc.definition.iter_markers(validity_marker_name)
+    ]
+    if not validity_markers:
+        return None
+    if len(validity_markers) > 1:
+        pytest.fail(f"'{test_name}': Too many '{validity_marker_name}' markers applied to test. ")
+    if len(validity_markers[0].args) == 0:
+        pytest.fail(f"'{test_name}': Missing fork argument with '{validity_marker_name}' marker. ")
+    if len(validity_markers[0].args) > 1:
+        pytest.fail(
+            f"'{test_name}': Too many arguments specified to '{validity_marker_name}' marker. "
+        )
+    fork_name = validity_markers[0].args[0]
+    if fork_name not in metafunc.config.fork_names:  # type: ignore
+        pytest.fail(
+            f"'{test_name}' specifies an invalid fork '{fork_name}' to the "
+            f"'{validity_marker_name}'. "
+        )
```

Comment:
```suggestion
        pytest.fail(
            f"'{test_name}' specifies an invalid fork '{fork_name}' to the "
            f"'{validity_marker_name}'. "
            f"List of valid forks: {', '.join(metafunc.config.fork_names)}"
        )
```
In case the user made a typo, could be useful to print the list of valid forks.